garden_ai/gardens.py

```diff
@@ -260,6 +218,185 @@ def expanded_json(self) -> JSON:
+        validate_assignment = True  # validators invoked on assignment
+        underscore_attrs_are_private = True
+
+
+class PublishedGarden(BaseModel):
+    """Metadata of a completed and published `Garden` object.
+
+    Attributes:
+        authors (list[str]):
+            The main researchers involved in producing the Garden. \
+            Personal name format should be: "Family, Given".
+
+        contributors (list[str]):
+            Acknowledge contributors to the development of this Garden. These\
+            should be distinct from `authors`.
+
+        title (str):
+            An official name or title for the Garden.
+
+        description (str):
+            A brief summary of the Garden and/or its purpose, to aid discovery by \
+            other Gardeners.
+
+        pipelines (list[RegisteredPipeline]):
+            List of the pipelines associated with this garden \
+            Note that these pipelines can also be accessed directly by their \
+            `short_name` (or alias) - see also `pipeline_names` attribute.
+
+        pipeline_names (list[str]):
+            Automatically generated list of python identifiers (i.e. variable names) \
+            usable for this garden's pipelines. Takes aliases into account.
+
+        pipeline_ids: List[str] = Field(default_factory=list)
+        pipeline_aliases: Dict[str, str] = Field(default_factory=dict)
+
+        doi (str):
+            A garden's DOI usable for citations.
+
+        version (str):
+            optional, defaults to "0.0.1".
+
+        language (str):
+            Recommended values are IETF BCP 47, ISO 639-1 language codes, such as:\
+            "en", "de", "fr". Defaults to "en".
+
+        tags (List[str]):
+            Tags, keywords, classification codes, or key phrases pertaining to the Garden.
+
+        year (str):
+            Defaults to current year. This attribute should be formatted 'YYYY'.
+
+    Examples:
+        Retrieving a remote Garden with a ``GardenClient`` instance::
+
+        ```python
+        gc = GardenClient()
+        garden = gc.get_published_garden("10.23677/placeholder")
+        ```
+    """
+
+    title: str = Field(...)
+    authors: List[str] = Field(...)
+    contributors: List[str] = Field(default_factory=list, unique_items=True)
+    doi: str = Field(...)
+    description: Optional[str] = Field(None)
+    publisher: str = "Garden-AI"
+    year: str = Field(default_factory=lambda: str(datetime.now().year))
+    language: str = "en"
+    tags: List[str] = Field(default_factory=list, unique_items=True)
+    version: str = "0.0.1"
+    pipelines: List[RegisteredPipeline] = Field(...)
+    pipeline_ids: List[str] = Field(...)
+    pipeline_aliases: Dict[str, str] = Field(default_factory=dict)
+    pipeline_names: List[str] = Field(...)
+    _env_vars: Dict[str, str] = PrivateAttr(default_factory=dict)
+
+    @root_validator(pre=True)
+    def verify_pipeline_integrity(cls, values):
```

Comment:
this smells odd -- why do `PublishedGardens` need `pipeline_ids` and `pipeline_names` as attributes when they already have full copies of their `RegisteredPipelines`? I think we shouldn't duplicate that state at the garden level when it's already available at the pipeline level. that way we don't need to handle them falling out of sync like this validator is anticipating